uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/preference/FuseServerRuntimePreferencePage.java

```diff
@@ -49,7 +51,11 @@ public FuseServerRuntimePreferencePage(ReferencedComposite referencedComposite)
 	 */
 	public void addServerRuntime(String type, String path) {
 		NewRuntimeWizardDialog dialog = addRuntime();
-		new DefaultTreeItem(SERVER_SECTION, type).select();
+		if (type.startsWith(SERVER_TYPE_KARAF_PREFIX)) {
```

Comment:
I suggest to add a new method parameter, e.g. 'category' so that we can avoid if-then-else structure. Or we could have a  parameter of type ServerBase. Note that the 'category' (here you use SERVER_SECTION) is already defined in classes ServerFuse and ServerKaraf.